src/Psl/Async/Scheduler.php

```diff
@@ -114,35 +115,43 @@ public static function defer(Closure $callback): string
     /**
      * Delay the execution of a callback.
      *
-     * @param float $delay The amount of time, to delay the execution for in seconds.
+     * @param DateTime\Duration|float $delay The amount of time, to delay the execution for in seconds.
```

Comment:
Would it make sense to deprecate the usage of `float` and drop it in next major release?
(same for all of those similar changes in this PR)

Reply:
I'm not sure, `Async\sleep(0.5)` IMO is still clear, while `Async\sleep(DateTime\Duration::milliseconds(500))` could be too verbose. some people might lean more toward using float for most cases.

Reply:
also, a lot of changes related to this in this PR are BC-breaking, so we need a new major for this PR, ( e.g parameter change on interfaces such as `IO\ReadHandleInterface`, and more )

Reply:
dropped float!